src/windows.rs

```diff
@@ -0,0 +1,463 @@
+                    ))];
+                };
+
+                // Get the Physical Monitors in the HMONITOR
+                let mut physical_number: u32 = 0;
+                if !BOOL(GetNumberOfPhysicalMonitorsFromHMONITOR(
+                    hmonitor,
+                    &mut physical_number,
+                ))
+                .as_bool()
+                {
+                    return vec![Err(SysError::ListingDevicesFailed(
+                        "GetNumberOfPhysicalMonitorsFromHMONITOR",
+                        GetLastError(),
+                    ))];
+                };
+                let mut physical_monitors = Vec::with_capacity(physical_number as usize);
+                if !BOOL(GetPhysicalMonitorsFromHMONITOR(
+                    &hmonitor,
+                    physical_number,
+                    physical_monitors.as_mut_ptr(),
+                ))
+                .as_bool()
+                {
+                    return vec![Err(SysError::ListingDevicesFailed(
+                        "GetPhysicalMonitorsFromHMONITOR",
+                        GetLastError(),
+                    ))];
+                }
+                physical_monitors.set_len(physical_number as usize);
+
+                // Get the Display Devices in the HMONITOR
+                let mut device_number = 0;
+                let mut device = DISPLAY_DEVICEW::default();
+                device.cb = std::mem::size_of::<DISPLAY_DEVICEW>() as u32;
+                let mut display_devices = Vec::new();
+                while EnumDisplayDevicesW(
+                    PWSTR(info.szDevice.as_mut_ptr()),
+                    device_number,
+                    &mut device,
+                    EDD_GET_DEVICE_INTERFACE_NAME,
+                )
+                .as_bool()
+                {
+                    device_number += 1;
+                    display_devices.push(device.clone());
+                }
+                if display_devices.len() != physical_monitors.len() {
+                    // There doesn't seem to be anyway to directly associate a physical monitor handle
+                    // with the equivalent display device, other than by array indexing
+                    // https://stackoverflow.com/questions/63095216/how-to-associate-physical-monitor-with-monitor-deviceid
+                    return vec![Err(SysError::EnumerationMismatch)];
+                }
+
+                physical_monitors
+                    .into_iter()
+                    .zip(display_devices)
+                    .map(|(physical_monitor, mut display_device)| {
+                        // Get a file handle for this physical monitor
+                        // Note this is a different type of handle
+                        let handle = CreateFileW(
+                            PWSTR(display_device.DeviceID.as_mut_ptr()),
+                            FILE_ACCESS_FLAGS(GENERIC_READ | GENERIC_WRITE),
+                            FILE_SHARE_MODE(0),
```

Comment:
Is it necessary to request exclusive access to the device? [The `DeviceIoControl` documentation](https://docs.microsoft.com/en-us/windows/win32/api/ioapiset/nf-ioapiset-deviceiocontrol) indicates that `FILE_SHARE_READ | FILE_SHARE_WRITE` should be used, which would play nicer with other applications.

Reply:
yep that works on my laptop